src/openclsim/critical_path/base_cp.py

```diff
@@ -66,9 +70,197 @@ def _make_recorded_activities_df(self):
         """
         Set a recorded_activity_df in self.
         Uses the logs of provided activities and sim objects, combines these, adds unique UUID
-        and reshape into format such that single row has a start time and an end time
+        and reshape into format such that single row has a start time and an end time.
         """
-        pass
+        # get all recorded log-events
+        all_recorded_events = self.combine_logs()
+
+        # reshape into set of activities with start, duration and end
+        recorded_activities_df = self.reshape_log(all_recorded_events)
+
+        # add unique identifier for activities (could be shared by multiple objects)
+        self.recorded_activities_df = self.add_unique_activity(recorded_activities_df)
+
+    def combine_logs(self):
+        """
+        Combines the logs of given (simulation) objects into a single pandas.Dataframe.
+        """
+        # check unique names - @Pieter maybe warning?
+        names = [obj.name for obj in self.object_list]
+        if not len(names) == len(set(names)):
+            raise ValueError("Names of your objects must be unique!")
+
+        # concat
+        log_all = pd.DataFrame()
+        for obj in self.object_list:
+            log = get_log_dataframe(obj)
+            log["SimulationObject"] = obj.name
+            log_all = pd.concat([log_all, log])
```

Comment:
changed the concat so that it is a single concat (and hence more efficient). However, the insert mutates (in place) and returns None so these proposed lines did not work. The offered alternative is not as concise, was not (quickly) able to think of 'oneliner'.

Reply:
You are here modifying the data frames by shallow reference. you cannot do that.

Reply:
Was quite ugly indeed.
New approach with double concat

```python

        log_list = [get_log_dataframe(obj) for obj in self.object_list]
        names_column = (
            pd.Series(names, name="SimulationObject")
            .repeat([len(df) for df in log_list])
            .reset_index(drop=True)
        )
        log_all = pd.concat(
            [names_column, pd.concat(log_list).reset_index(drop=True)], axis=1
        )

```